src/auto-evo/RunResults.cs

```diff
@@ -811,6 +811,151 @@ void OutputPopulationForPatch(Species species, Patch patch, long population)
             return builder;
         }
 
+        public void LogResultsToTimeline(GameWorld world, List<ExternalEffect> effects = null)
+        {
+            var newSpecies = GetNewSpecies();
+
+            foreach (var patch in world.Map.Patches.Values)
+            {
+                foreach (var species in patch.SpeciesInPatch.Keys)
+                {
+                    var globalPopulation = GetGlobalPopulation(species, true, true);
+                    var previousGlobalPopulation = world.Map.GetSpeciesGlobalPopulation(species);
+
+                    var finalPatchPopulation = GetPopulationInPatch(species, patch);
+                    var previousPatchPopulation = patch.GetSpeciesPopulation(species);
+
+                    finalPatchPopulation += CountSpeciesSpreadPopulation(species, patch);
+
+                    if (results[species].SplitOffPatches?.Contains(patch) == true)
+                    {
+                        // All population splits off
+                        finalPatchPopulation = 0;
+                    }
+
+                    // Apply external effects
+                    if (effects != null && world.Map.CurrentPatch.ID == patch.ID)
```

Comment:
Just for future note, @FuJa0815's PR needs to adjust this part. I'm really hoping I can commit to getting the dynamic MP and patchwide population system merged early for 0.5.8 this time.